cvat/apps/engine/migrations/0084_honeypot_support.py

```diff
@@ -0,0 +1,171 @@
+# Generated by Django 4.2.15 on 2024-09-23 13:11
+
+from typing import Collection
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import cvat.apps.engine.models
+
+
+def get_frame_step(db_data) -> int:
+    v = db_data.frame_filter or "step=1"
+    return int(v.split("=")[-1])
+
+
+def get_rel_frame(abs_frame: int, db_data) -> int:
+    data_start_frame = db_data.start_frame
+    step = get_frame_step(db_data)
+    return (abs_frame - data_start_frame) // step
+
+
+def get_segment_rel_frame_set(db_segment) -> Collection[int]:
+    db_data = db_segment.task.data
+    data_start_frame = db_data.start_frame
+    data_stop_frame = db_data.stop_frame
+    step = get_frame_step(db_data)
+    frame_range = range(
+        data_start_frame + db_segment.start_frame * step,
+        min(data_start_frame + db_segment.stop_frame * step, data_stop_frame) + step,
+        step,
+    )
+
+    if db_segment.type == "range":
+        frame_set = frame_range
+    elif db_segment.type == "specific_frames":
+        frame_set = set(frame_range).intersection(db_segment.frames or [])
+    else:
+        raise ValueError(f"Unknown segment type: {db_segment.type}")
```

Comment:
I am not sure that raising uncaught exception is good in migration file

Reply:
On our prod we only have `specific_frames` defined, so, it will not be a problem

Reply:
The reason is to fail the migration, if the DB contains invalid entries. We don't know why they are there and what to do with them.